File: 108-ConvertSortedArrayToBinarySearchTree/108-ConvertSortedArrayToBinarySearchTree.cpp
// Last updated: 13/07/2026, 22:27:06
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return helper(nums, 0, nums.size() - 1);
    }

    TreeNode* helper(const vector<int>& nums, int left, int right){
        if (left > right){
            return nullptr;
        }
        int mid = left + (right - left) / 2;
        return new TreeNode(
            nums[mid],
            helper(nums, left, mid -1),
            helper(nums, mid + 1, right)
        );
    }
};

File: 108-ConvertSortedArrayToBinarySearchTree/108-ConvertSortedArrayToBinarySearchTree.py
# Last updated: 13/07/2026, 22:27:19
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        total = len(nums)
        if (not total):
            return None
        
        mid = total // 2
        return TreeNode(
            nums[mid], 
            self.sortedArrayToBST(nums[ : mid]), 
            self.sortedArrayToBST(nums[mid + 1 : ])
        )